src/main/java/com/statful/sender/MetricsHolder.java

```diff
@@ -64,26 +71,40 @@ public final boolean addMetric(final DataPoint dataPoint) {
         }
 
         boolean inserted = this.buffer.offer(dataPoint);
+
+        this.shouldFlush(inserted);
+
         if (!inserted) {
             LOGGER.warn("metric could not be added to buffer, discarding it {} ", dataPoint.toMetricLine());
```

Comment:
Eventually we could retry after flushing the buffers instead of discarding but good enough for now.